beanstalk_worker/worker_start_scan.py

```diff
@@ -101,6 +102,35 @@ def run_a_scanner(self, scanner, image_under_test):
         # if scanner failed to run, this will return {}, do check at receiver end
         return json_data
 
+    def export_scanner_logs(self, scanner, data):
+        """
+        Export scanner logs in given directory
+        """
+        logs_file_path = os.path.join(
+                self.job_info["logs_dir"],
+                constants.SCANNERS_RESULTFILE[scanner][0])
+        logger.log(
+                level=logging.INFO,
+                msg="Scanner=%s result log file:%s" % (scanner, logs_file_path)
+                )
+        try:
+            fin = open(logs_file_path, "w")
+        except IOError as e:
+            logger.log(
+                level=logging.CRITICAL,
+                msg="Failed to write scanner=%s logs on NFS share." % scanner)
```

Comment:
@navidshaikh let me give you a scenario.

- we're using an NFS share where we write data
- we open a file to write data to it
- connection to NFS share got broken before we could actually write to the file

What would happen in this situation? Are we going to be able to write to the file? 

Having `json.dump` in the `else` block is going to crash this program if I understand correctly.

So, in this situation, you could open the file for writing but couldn't write to it. Two different things.

@kbsingh @arrfab @bamachrn thoughts on the scenario I mentioned and its consequence? Please correct me if I'm wrong.